sfn/src/test/scala/uk/gov/nationalarchives/DASFNClientTest.scala

```diff
@@ -3,66 +3,183 @@ package uk.gov.nationalarchives
-    request.stateMachineArn should be(arn)
+    request.sfnArn should be(arn)
     request.input should be("""{"message":"testMessage","value":"testValue"}""")
   }
 
   "startExecution" should "return an error if there is an error from the SFN API" in {
-    val sfnAsyncClient = mock[SfnAsyncClient]
-    when(sfnAsyncClient.startExecution(any[StartExecutionRequest]))
-      .thenThrow(new RuntimeException("Error starting execution"))
+    val sfnExecutions = ListBuffer[SfnExecutions]()
+    val sfnAsyncClient = createClient(sfnExecutions, errors = Option(Errors(startExecution = true)))
 
     val client = DASFNClient[IO](sfnAsyncClient)
-
     val ex = intercept[Exception] {
-      client.startExecution(arn, TestInput("testMessage", "testValue"), Option("testName")).unsafeRunSync()
+      client.startExecution(arn, TestInput("testMessage", "testValue"), Option("testName")).attempt.unsafeRunSync()
     }
+
     ex.getMessage should equal("Error starting execution")
   }
+
+  "listExecutions" should "list executions with the correct status and arn" in {
+    val initialExecutions = ListBuffer(
+      SfnExecutions("name1running", "arn1", "", "running"),
+      SfnExecutions("name2failed", "arn2", "", "failed"),
+      SfnExecutions("name2running", "arn2", "", "running")
+    )
+    val sfnAsyncClient = createClient(initialExecutions)
+    val client = DASFNClient[IO](sfnAsyncClient)
+
+    val runningArn2 = client.listStepFunctions("arn2", Running).unsafeRunSync()
+    val failedArn2 = client.listStepFunctions("arn2", Failed).unsafeRunSync()
+    val runningArn1 = client.listStepFunctions("arn1", Running).unsafeRunSync()
+
+    runningArn2.size should equal(1)
+    runningArn1.size should equal(1)
+    failedArn2.size should equal(1)
+
+    runningArn2.head should equal("name2running")
+    runningArn1.head should equal("name1running")
+    failedArn2.head should equal("name2failed")
+  }
+
+  "listExecutions" should "return an error if there is an error from AWS" in {
+    val initialExecutions = ListBuffer[SfnExecutions]()
+    val sfnAsyncClient = createClient(initialExecutions, errors = Option(Errors(listExecutions = true)))
+    val client = DASFNClient[IO](sfnAsyncClient)
+
+    val ex = intercept[Exception] {
+      client.listStepFunctions("arn2", Running).attempt.unsafeRunSync()
+    }
+
+    ex.getMessage should equal("Error listing executions")
+  }
+
+  "sendTaskSuccess" should "return no errors if the task token is valid" in {
+    val initialExecutions = ListBuffer(
+      SfnExecutions("name1running", "arn1", "", "running", "taskToken")
+    )
+    val sfnAsyncClient = createClient(initialExecutions)
+    val client = DASFNClient[IO](sfnAsyncClient)
+
+    client.sendTaskSuccess("taskToken").unsafeRunSync()
+  }
+
+  "sendTaskSuccess" should "return an error if the task token is invalid" in {
+    val initialExecutions = ListBuffer(
+      SfnExecutions("name1running", "arn1", "", "running", "taskToken")
```

Comment:
1. Maybe `taskToken` could be `invalidTaskToken`
2. Maybe line 170 could be `client.sendTaskSuccess( initialExecutions.taskToken).unsafeRunSync()`?

Reply:
This is testing that when we send a task token that's incorrect, we get an exception. So the task token on line 164 is a valid running task token and the one on line 170 is the invalid one which causes the exception. They shouldn't be the same.